### DISCUSSION TITLE
"Slow" navigation in menu

### DISCUSSION THREAD

**Discussion by chrinaf** [davetcc/tcMenu#428] [Q&A]
I am trying to create my first project with an ESP32-C3 and Arduino IDE. I am using a DFRobot Beetle ESP32 - C3.
Creating and displaying a menu with TcMenuDesigner works so far. Thanks a lot for the great tool!
For navigation I use an EC11 rotary encoder. It works, but pretty slow. I get maximum of 2 inputs per second, no matter how fast I turn the encoder. Screen updates per seconds is set to 20.

I have the same issue when I use the "buttonRotaryEncoder" example. The LED turns on and off max. 2 times per second.
I could solve it when setting accelerationMode = HWACCEL_NONE. Then it works significantly faster.

However, I did not find a way to do this in my sketch. It might be a stupid question, but how can I change the accelerationMode without defining a new rotary encoder? Needless to say, that I am a beginner and using tcMenu as a way to get insights into a more complex project.

**Comment by davetcc** [comment]
Apologies for the slow response, I would suggest that you first try using the IoAbstraction rotary encoder example sketch first - IE test in isolation, and see if that has the same results. 

It is always easiest to distill down the smallest sketch possible to track down issues.

**Reply by chrinaf** [reply]
Thanks for the reply Dave!
I used buttonRotaryEncoder.ino and have the same slow reaction. 
When `accelerationMode = HWACCEL_NONE`, it works significantly faster.
That would solve my issues. However, I dont know how to set the accelerationMode in my sketch. 

My sketch is the output of the TcMenuDesigner, without any changes. The rotary encoder is not defined in the sketch and I don't know how to change values such as acceleration mode. I tried to use the `void encoderChanged() override;` in SwitchInput.h but with no success. 

My Sketch:

```
void setup() {
  setupMenu();
  void showWorkout(unsigned int encoderValue, RenderPressMode userClicked);
  menuMgr.setUseWrapAroundEncoder(true);
  
}

void loop() {
  taskManager.runLoop();
}

void CALLBACK_FUNCTION startWorkout(int id) {
  renderer.takeOverDisplay(showWorkout);
}
```
